sup3r/models/conditional_moments.py

```diff
@@ -0,0 +1,850 @@
+        """Load the model with its sub-networks from a previously saved-to
+        output directory.
+
+        Parameters
+        ----------
+        model_dir : str
+            Directory to load model files from.
+        verbose : bool
+            Flag to log information about the loaded model.
+
+        Returns
+        -------
+        out : BaseModel
+            Returns a pretrained gan model that was previously saved to out_dir
+        """
+        if verbose:
+            logger.info('Loading model from disk in directory: {}'
+                        .format(model_dir))
+            msg = ('Active python environment versions: \n{}'
+                   .format(pprint.pformat(VERSION_RECORD, indent=4)))
+            logger.info(msg)
+
+        fp_gen = os.path.join(model_dir, 'model_gen.pkl')
+        params = cls.load_saved_params(model_dir, verbose=verbose)
+        return cls(fp_gen, **params)
+
+    def load_network(self, model, name):
+        """Load a CustomNetwork object from hidden layers config, .json file
+        config, or .pkl file saved pre-trained model.
+
+        Parameters
+        ----------
+        model : str | dict
+            Model hidden layers config, a .json with "hidden_layers" key, or a
+            .pkl for a saved pre-trained model.
+        name : str
+            Name of the model to be loaded
+
+        Returns
+        -------
+        model : phygnn.CustomNetwork
+            CustomNetwork object initialized from the model input.
+        """
+
+        if isinstance(model, str) and model.endswith('.json'):
+            model = safe_json_load(model)
+            self._meta[f'config_{name}'] = model
+            if 'hidden_layers' in model:
+                model = model['hidden_layers']
+            elif ('meta' in model
+                  and f'config_{name}' in model['meta']
+                  and 'hidden_layers' in model['meta'][f'config_{name}']):
+                model = model['meta'][f'config_{name}']['hidden_layers']
+            else:
+                msg = ('Could not load model from json config, need '
+                       '"hidden_layers" key or '
+                       f'"meta/config_{name}/hidden_layers" '
+                       ' at top level but only found: {}'
+                       .format(model.keys()))
+                logger.error(msg)
+                raise KeyError(msg)
+
+        elif isinstance(model, str) and model.endswith('.pkl'):
+            model = CustomNetwork.load(model)
+
+        if isinstance(model, list):
+            model = CustomNetwork(hidden_layers=model, name=name)
+
+        if not isinstance(model, CustomNetwork):
+            msg = ('Something went wrong. Tried to load a custom network '
+                   'but ended up with a model of type "{}"'
+                   .format(type(model)))
+            logger.error(msg)
+            raise TypeError(msg)
+
+        return model
+
+    @property
+    def means(self):
```

Comment:
I reorganized `abstract`, `conditionalMoment` and `base`. Most duplicate code is gone now.  I did leave `meta` since it is an abstract method which needs to be redefined. `model_params` change a little bit between Sup3rCondMom and Sup3rGan, so I left it as well. All the normalization stuff is in the abstractSingleModel now. I think it also makes sense to do this since any future model will presumably need these normalization methods. I'll leave this unresolved for now until you validate.

Reply:
Nice looks great and pylint says no duplicate code now. Thanks!